extensions/applicationinsights-properties-js/src/Context/User.ts

```diff
@@ -172,9 +182,12 @@ export class User implements IUserContext {
             };
 
             _self.update = (userId?: string) => {
-                let user_id = userId ? userId : _generateNewId();
-                let user_cookie = _generateNewCookie(user_id);
-                _setUserCookie(user_cookie.join(User.cookieSeparator));
+                // Optimizations to avoid setting and processing the cookie when not needed
+                if (_self.id !== userId || !_self.isUserCookieSet) {
```

Comment:
Add an extra performance based check so we don't try and set the cookie if
- The value was already set
- The new userId is the same as the previous one.